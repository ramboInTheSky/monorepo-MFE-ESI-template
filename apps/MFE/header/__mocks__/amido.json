{
  "id": "33d0153c-ed8b-4a65-b63c-5ad5edddd187",
  "name": "Default",
  "realm": "amido",
  "territory": "GB",
  "language": "en",
  "regions": [
    {
      "type": "Brand",
      "title": "Brand",
      "accessibilityTitle": "Brand",
      "elements": [
        {
          "elements": [],
          "type": "link",
          "url": "/",
          "openInNewWindow": false,
          "icon": "/icons/header/brand/amido/logo.svg",
          "text": "Amido Logo",
          "accessibilityText": "Amido Icon",
          "tooltip": "Amido Icon",
          "wideModeIcon": "/icons/header/brand/amido/logo.svg",
          "narrowModeIcon": "/icons/header/brand/amido/logo.svg"
        }
      ]
    },
    {
      "type": "QuickLinks",
      "title": "QuickLinks",
      "accessibilityTitle": "QuickLinks",
      "elements": [
        {
          "elements": [
            {
              "type": "MyAccountLoggedIn",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "<username>",
              "accessibilityText": "<username>",
              "tooltip": "<username>'s account",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            },
            {
              "type": "MyAccountLoggedOut",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "My Account",
              "accessibilityText": "Sign in to view account details",
              "tooltip": "Sign in to view account details",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            }
          ],
          "type": "MyAccount",
          "url": null,
          "openInNewWindow": true,
          "icon": null,
          "text": "MyAccount",
          "accessibilityText": "MyAccount",
          "tooltip": null,
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/help",
          "openInNewWindow": true,
          "icon": null,
          "text": "Help",
          "accessibilityText": "Help",
          "tooltip": "Help - FAQ's, how to shop online, sizing guides and more",
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/stores/",
          "openInNewWindow": false,
          "icon": null,
          "text": "Store Locator",
          "accessibilityText": "Store Locator",
          "tooltip": "Find your nearest Amido Store",
          "wideModeIcon": null,
          "narrowModeIcon": null
        }
      ]
    }
  ]
}
