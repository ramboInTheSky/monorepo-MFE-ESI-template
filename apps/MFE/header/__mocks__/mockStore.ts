import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import {mockColors} from "@monorepo/themes"
import {State} from "../src/ducks"
import apiData from "./amido.json"
import {ShoppingBag, ChargesAndIncentive, Item} from "../src/models/shoppingbag"

export const mockStyles = {
    sm: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xs: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    md: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        margin: "10rem",
        decoration: "none",
    },
    lg: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xl: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
}

export const mockMyAccountStyles = {
    ...mockStyles,
    SmallIcon: mockStyles,
    SmallIconContainer: mockStyles,
    IconOnly: mockStyles,
    IconContainer: mockStyles,
    ToolTip: mockStyles,
    Divider: mockStyles,
}

export const mockCountrySelectorStyles = {
    ...mockStyles,
    Container: mockStyles,
}

const mockShoppingBagStyles = {
    ...mockStyles,
    productImage: mockStyles,
    itemsContainer: mockStyles,
}

const mockFavouritesStyles = {
    ...mockStyles,
    img: mockStyles,
}

export const mockRealmStyles = {
    CTA: mockStyles,
    UpperHeader: mockStyles,
    Meganav: mockStyles,
    PrimaryNav: mockStyles,
    Brand: mockStyles,
    SearchBox: mockStyles,
    Checkout: mockStyles,
    Quicklinks: mockStyles,
    SearchBoxDivider: mockStyles,
    SearchBoxIcon: mockStyles,
    Favourites: mockFavouritesStyles,
    CountrySelector: mockCountrySelectorStyles,
    ShoppingBag: mockShoppingBagStyles,
    MyAccount: mockMyAccountStyles,
    AutoCompleteButton: mockStyles,
    DrawerContainer: mockStyles,
}

const discount = 10
export const mockShoppingBag = {
    itemCount: 2,
    loading: false,
    bag: {
        FirstName: "cypress-account",
        Description: "Excluding UK Standard Delivery (Normally £3.99)",
        MultiBuyDiscount: 1,
        FinalOrderValue: 89,
        FinalOrderValueFormatted: "£89.00",
        MultiBuyDiscountFormatted: "£1.00",
        ChargesAndIncentives: [
            {
                AdditionalAmount: -discount,
                AdditionalAmountFormatted: `-£${discount.toFixed(2)}`,
                AdditionalCode: "UKCO1",
                MaximumOrderValue: 0,
                MinimumOrderValue: 15,
                OfferShortDescription: "Welcome Offer",
                OfferTypeDescription: "Welcome Offer",
                Type: "PRO",
            } as ChargesAndIncentive,
        ],
        ItemCount: 2,
        Items: [
            {
                ItemID: 1,
                ItemNumber: "436441",
                OptionNo: "04",
                SizeDescription: "10 S",
                Price: "45.00",
                Quantity: 10,
                StockMessage: "Delayed",
                Description: "Black Lift, Slim And Shape Skinny Jeans",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans",
                CistStoreName: "",
                Url: "/path1",
            } as Item,
            {
                ItemID: 2,
                ItemNumber: "436642",
                OptionNo: "04",
                SizeDescription: "15 L",
                Price: "45.00",
                Quantity: 5,
                StockMessage: "In Stock",
                Description: "Black Lift, Slim And Shape Skinny Jeans 2",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans 2",
                CistStoreName: "LEICESTER - HIGH CROSS SHOPPING CENTRE",
                Url: "/path2",
            } as Item,
        ],
    } as ShoppingBag,
}

export const mockRecents = {
    queryIds: ["celio", "gucci", "prada", "&", "#", "+"],
}

export const mockAutoComplete: any = {
    isLoading: false,
    q: "socks",
    suggestions: [
        {
            dq: "girls socks",
            q: "girls socks",
        },
        {
            dq: "boys socks",
            q: "boys socks",
        },
    ],
    products: [],
    numFound: 2,
    parameters: {
        accountId: "abc",
        domainKey: "Amido",
    },
}

export const mockLanguages = {
    currentLanguageText: "English",
    altLanguageName: "ar",
    altLanguageUrl: "/ar",
    currentLanguageName: "en",
    direction: "ltr",
    siteUrl: "http://amido.com",
}

export const mockSearch = {
    typedCharacters: "",
    showAutocomplete: false,
    showRecentSearch: false,
}

export const mockText = {
    recentSearches: {
        enriched: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "Your recent searches: ",
        },
        simple: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "RECENT SEARCHES",
        },
    },
    searchBox: {
        label: "Search",
        bigPlaceholder: "Search product",
        smallPlaceholder: "Search...",
        clearButton: {
            alt: "Clear search button",
        },
        closeButton: {
            text: "Close",
        },
    },
    myAccount: {
        title: "View Account Summary",
        buttonText: "SIGN OUT",
        altText: "My account icon",
        tooltipAltText: "icon",
        notText: "Not",
    },
    checkout: {
        text: "Checkout",
    },
    autoComplete: {
        loadingText: "Loading...",
        noResults: "No results found for ",
        headerText: "TOP MATCHES",
        searchLinktext: "See all results",
        enrich: {
            termTitle: "Top results for ",
        },
    },
    drawer: {
        closeIconText: "Close icon",
    },
    favourites: {
        altText: "Favourites icon",
    },
    miniShoppingBag: {
        emptyBagText: "Your shopping bag is empty",
        totalText: "Total",
        promotionalText: "Promotional Savings",
        viewAndEditBag: "VIEW/EDIT BAG",
        sizeText: "Size:",
        qtyText: "Qty:",
        at: "at",
        negative: "-",
        itemText: "Item",
        itemsText: "Items",
        inBagText: "In Bag",
        salePrice: "Sale Price",
    },
    shoppingBag: {
        altText: "Shopping bag icon",
        youhaveText: "You have",
        itemText: "item",
        itemsText: "items",
        inYourBagText: "in your bag",
    },
    countryLangSelector: {
        headerTitle: "CHANGE COUNTRY",
        locationText: "Location",
        languageText: "Language",
        closeText: "Close",
        stayText: "Stay on Amido.co.uk",
        shopNowText: "Shop now",
        ROWText: "Can't find the country you're looking for?",
        ROWLinkText: "View All Countries",
    },
    countryChangeModal: {
        confirm: "CONTINUE",
        cancel: "CANCEL",
        bodyTextPt1: "Are you sure you want to navigate away from this site?",
        bodyTextPt2: "If you navigate away from this site you will leave your shopping bag and it’s contents.",
        titleText: "Confirm country change",
    },
    cookiePolicy: {
        title: "Cookie Policy",
        body: "We use cookies to provide you with the best posible experience. By continuing to use our site, you agree to our use of cookies.",
        link: "Find out more",
        linkText: "about managing your cookie settings.",
        closeIconAltText: "Close",
    },
    saleBagWarning: {
        title: "Careful",
        textOne: "You could lose your sale bag.",
        textTwo: "Would you like to return to our Sale Site?",
        primaryButtonText: "YES, RETURN TO THE SALE",
        secondaryButtonText: "NO, THANK YOU",
    },
    skipToMainContent: "Skip to main content",
    closeIconAltText: "Close",
}

export const mockState: State = {
    settings: {
        "userConsentManagement.enabled": false,
        variant: "default",
    },
    user: {firstName: "", loggedIn: false, userUpdated: false},
    autocomplete: mockAutoComplete,
    recents: mockRecents,
    shoppingBag: mockShoppingBag,
    search: mockSearch,
    template: "default",
    request: {
        headers: {"x-monorepo-territory": "gb"},
        url: "/",
        siteUrl: "fakeamido.com",
        timeMachineDate: "20201215",
        isInternationalCountry: false,
        fullTerritoryName: "United Kingdom",
        geolocationBaseUrl: "",
        geolocationVersion: 0,
        bloomReachCachingCookieList: "",
        bloomReachCachingEnabled: false,
        useTimeMachineCookie: false,
        currencyCode: "EUR",
        bloomreachGroupLocation: "southernhemisphere",
        bloomreachDomainKey: "GB",
        monetateSDK: false,
        accountMonetateSDK: "",
        cookieDomain: ".amido.com",
        showSaleWarningBag: false,
    },
    languages: mockLanguages,
    data: {...apiData, language: ""},
    textAlignment: "ltr",
    text: mockText,
    features: {
        SearchBar: {
            Value: "SimpleSearch",
            RecentSearch: {
                MaxItems: 12,
            },
            Autocomplete: {
                MaxItems: 10,
                ProductsMaxItems: {
                    xs: 8,
                    sm: 8,
                    md: 4,
                    lg: 4,
                    xl: 5,
                },
            },
        },
    },
    favourites: {enableFavourites: true, hasFavourites: true},
    countrySelector: {
        showCountrySelector: false,
        isActive: true,
        countriesList: null,
        selectedCountry: null,
        selectedLanguage: null,
        defaultCountryCode: null,
        showOverlay: true,
        showBFPOFlag: false,
        loaded: false,
        requestedCountryChange: false,
        countrySelectorEndpoint: "",
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme = {styles: mockRealmStyles, colours: mockColors, direction: "ltr"}
export const mockConfigureStore = configureMockStore([thunk])
const mockStore = mockConfigureStore(mockState)

export default mockStore
