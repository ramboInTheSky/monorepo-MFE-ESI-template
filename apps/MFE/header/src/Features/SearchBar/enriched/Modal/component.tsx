import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

/*
 * Title: Using to style top !important
 * problem: we do not want to use !important but we want a modal and a modal is expected to cover the entire screen
 * in this case however, material UI probably assumes you'll have the entire screen covered and uses inline styles
 * to define this behaviour. But here the modal as it is the case of next, partially exposes the header area hence it
 * does not fully cover the entire screen.
 * Solution: for now we have to use !important
 * Refs:
 *    https://github.com/mui-org/material-ui/issues/16441
 *    https://github.com/mui-org/material-ui/issues/16442
 */

export const Wrapper = styled.div`
    & > div {
        transition: opacity 0.2s ease;
        animation-delay: 250ms;

        & > div:first-child {
            position: fixed;
            top: 0;
            background: rgba(0, 0, 0, 0.9);
        }

        @media (min-width: ${breakpoints.values.md}px) {
            top: ${props => props.theme.styles.UpperHeader.md.height} !important;
        }

        @media (min-width: ${breakpoints.values.lg}px) {
            top: ${props => props.theme.styles.UpperHeader.lg.height} !important;
        }
    }
`
