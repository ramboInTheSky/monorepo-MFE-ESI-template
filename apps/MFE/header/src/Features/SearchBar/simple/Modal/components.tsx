import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

/*
 * Title: Using to style top !important
 * problem: we do not want to use !important but we want a modal and a modal is expected to cover the entire screen
 * in this case however, material UI probably assumes you'll have the entire screen covered and uses inline styles
 * to define this behaviour. But here the modal as it is the case of next, partially exposes the header area hence it
 * does not fully cover the entire screen.
 * Solution: for now we have to use !important
 * Refs:
 *    https://github.com/mui-org/material-ui/issues/16441
 *    https://github.com/mui-org/material-ui/issues/16442
 */

export const Wrapper = styled.div`
    & > div {
        position: absolute !important;
        right: auto !important;
        left: auto !important;
        top: 3rem !important;
        bottom: auto !important;

        & > div:first-child {
            top: 3rem;
            background: rgba(0, 0, 0, 0.9);
            opacity: 0 !important;
        }

        @media (min-width: ${breakpoints.values.md}px) {
            & > div:first-child {
                top: ${props => props.theme.styles.UpperHeader.md.height} !important;
            }
        }

        @media (min-width: ${breakpoints.values.lg}px) {
            & > div:first-child {
                top: ${props => props.theme.styles.UpperHeader.lg.height} !important;
            }
        }
    }
`
