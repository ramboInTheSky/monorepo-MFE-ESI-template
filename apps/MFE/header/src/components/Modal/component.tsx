import React from "react"
import {Modal, ModalProps} from "@mui/material"
import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"
import {ViewportStyles} from "models/regions"

type Props = ModalProps & {
    styles: ViewportStyles
}

/*
 * Title: Using to style top !important
 * problem: we do not want to use !important but we want a modal and a modal is expected to cover the entire screen
 * in this case however, material UI probably assumes you'll have the entire screen covered and uses inline styles
 * to define this behaviour.
 * Solution: for now we have to use !important
 * Refs:
 *    https://github.com/mui-org/material-ui/issues/16441
 *    https://github.com/mui-org/material-ui/issues/16442
 */

export const Container = styled(({...rest}) => <Modal {...rest} />)<Props>`
    transition: opacity 0.2s ease;
    animation-delay: 250ms;

    & > div:first-child {
        position: fixed;
    }

    && > div:first-child {
        background: ${props =>
            `${
                props.$darkerBackdrop
                    ? `rgba(${props.theme.colours.palette.modal.backdrop.secondary})`
                    : `rgba(${props.theme.colours.palette.modal.backdrop.primary})`
            }`};
    }

    @media (min-width: ${breakpoints.values.md}px) {
        & > div:first-child {
            top: ${props => props.theme.styles.UpperHeader.md.height} !important;
        }
    }

    @media (min-width: ${breakpoints.values.lg}px) {
        & > div:first-child {
            top: ${props => props.theme.styles.UpperHeader.lg.height} !important;
        }
    }
`
