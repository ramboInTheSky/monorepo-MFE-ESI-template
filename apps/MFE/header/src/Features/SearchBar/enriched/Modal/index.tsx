import React from "react"

import {Wrapper} from "./component"

import Modal from "../../../../components/Modal"

export type ModalProps = {
    open: boolean
    closeText: string
    handleClose: () => void
    children: JSX.Element
}
const EnrichModal = ({open, handleClose, children, closeText}: ModalProps) => (
    <Wrapper>
        <Modal open={open} handleClose={handleClose} hideBackdrop={false}>
            <>
                {children}
                <div tabIndex={0} role="button" onFocus={handleClose} aria-label={closeText} />
            </>
        </Modal>
    </Wrapper>
)

export default EnrichModal
