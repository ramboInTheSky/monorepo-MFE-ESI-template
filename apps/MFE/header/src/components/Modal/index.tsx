import React from "react"
import {Backdrop} from "@mui/material"
import {Container} from "./component"
import {MODAL_TIMEOUT} from "../../config/constants"
import {handlePropagation} from "../../utils/handlePropagation"

export type ModalProps = {
    open: boolean
    handleClose: () => void
    children: JSX.Element
    darkerBackdrop?: boolean
    hideBackdrop?: boolean
}
const innerModalDefaultProps = {
    disableRestoreFocus: true,
    disableAutoFocus: true,
    disableEnforceFocus: true,
    disablePortal: true,
    keepMounted: false,
    disableScrollLock: true,
}

const Modal = ({open, handleClose, children, darkerBackdrop, hideBackdrop = false}: ModalProps) => (
    <Container
        {...innerModalDefaultProps}
        data-testid="modal"
        open={open}
        onClose={handleClose}
        onClick={handlePropagation}
        closeAfterTransition
        BackdropComponent={Backdrop}
        BackdropProps={{timeout: MODAL_TIMEOUT}}
        $darkerBackdrop={darkerBackdrop}
        hideBackdrop={hideBackdrop}
    >
        <>{children}</>
    </Container>
)

export default Modal
