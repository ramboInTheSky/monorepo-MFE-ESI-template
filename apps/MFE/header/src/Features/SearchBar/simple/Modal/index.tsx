import React from "react"

import {Wrapper} from "./components"

import Modal from "../../../../components/Modal"

export type ModalProps = {
    open: boolean
    closeText: string
    handleClose: () => void
    children: JSX.Element
}

const SimpleModal = ({open, handleClose, children, closeText}: ModalProps) => {
    return (
        <Wrapper>
            <Modal open={open} handleClose={handleClose} hideBackdrop>
                <>
                    {children}
                    <div tabIndex={0} role="button" onFocus={handleClose} aria-label={closeText} />
                </>
            </Modal>
        </Wrapper>
    )
}

export default SimpleModal
