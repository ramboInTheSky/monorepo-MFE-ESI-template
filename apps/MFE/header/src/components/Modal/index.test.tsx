import React from "react"
import {render, fireEvent, screen} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "../../../__mocks__/mockStore"
import Modal, {ModalProps} from "."

const Fake = () => <div> fake </div>

jest.mock("../../utils/handlePropagation", () => {
    return {
        handlePropagation: jest.fn(),
    }
})

describe("Components - Modal: ", () => {
    let props: Omit<ModalProps, "children">
    beforeEach(() => {
        props = {
            open: true,
            handleClose: jest.fn(),
        }
    })
    it("should match the snapshot", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <Modal {...props}>
                    <Fake />
                </Modal>
            </SCThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it("should match the snapshot when hideBackdrop is true", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <Modal {...props} hideBackdrop>
                    <Fake />
                </Modal>
            </SCThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it("Should be visible when open state is true", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <Modal {...props}>
                    <Fake />
                </Modal>
            </SCThemeProvider>,
        )
        const got = screen.getByTestId("modal")
        expect(got).toHaveStyle("visibility:visible")
        expect(got).toContainHTML("<div> fake </div>")
    })

    it("Should call handleClose function when backdrop is clicked", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <Modal {...props}>
                    <Fake />
                </Modal>
                ,
            </SCThemeProvider>,
        )
        const backdrop = screen.getByTestId("modal").firstChild as HTMLElement
        fireEvent.click(backdrop)
        expect(props.handleClose).toHaveBeenCalledTimes(1)
    })
})
