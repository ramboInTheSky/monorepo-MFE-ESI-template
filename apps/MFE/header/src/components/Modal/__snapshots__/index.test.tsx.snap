// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Modal:  should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition: opacity 0.2s ease;
  transition: opacity 0.2s ease;
  -webkit-animation-delay: 250ms;
  animation-delay: 250ms;
}

.c0 > div:first-child {
  position: fixed;
}

.c0.c0 > div:first-child {
  background: rgba(0,0,0,0.5);
}

@media (min-width:768px) {
  .c0 > div:first-child {
    top: 2.75rem !important;
  }
}

@media (min-width:1024px) {
  .c0 > div:first-child {
    top: 2.75rem !important;
  }
}

<div
    class="c0"
    data-testid="modal"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div>
       fake 
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`Components - Modal:  should match the snapshot when hideBackdrop is true 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition: opacity 0.2s ease;
  transition: opacity 0.2s ease;
  -webkit-animation-delay: 250ms;
  animation-delay: 250ms;
}

.c0 > div:first-child {
  position: fixed;
}

.c0.c0 > div:first-child {
  background: rgba(0,0,0,0.5);
}

@media (min-width:768px) {
  .c0 > div:first-child {
    top: 2.75rem !important;
  }
}

@media (min-width:1024px) {
  .c0 > div:first-child {
    top: 2.75rem !important;
  }
}

<div
    class="c0"
    data-testid="modal"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div>
       fake 
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
